volmdlr/wires.py

```diff
@@ -188,6 +188,49 @@ def sort_points_along_wire(self, points):
 
         return sorted(points, key=lambda point: self.abscissa(point))
 
+    def is_ordered(self, tol=1e-6):
+        '''
+        check if the wire's primitives are ordered or not
+        '''
+
+        for primitive_1, primitive_2 in zip(self.primitives, self.primitives[1:]):
+            if primitive_1.end.point_distance(primitive_2.start) > tol:
+                return False
+        return True
+
+    def order_wire(self, tol=1e-6):
+        '''
+        order wire's primitives
+        '''
+
+        if self.is_ordered(tol=tol):
+            return self.copy()
+
+        ordered_wire = self.copy()
+
+        new_primitives = [ordered_wire.primitives[0]]
+        primitives = ordered_wire.primitives[1:]
+        length_primitives = len(primitives) + 1
+
+        while len(new_primitives) < length_primitives:
+            for primitive in primitives:
+                if new_primitives[0].start.point_distance(primitive.start) < tol:
+                    new_primitives.insert(0, primitive.reverse())
+                    primitives.remove(primitive)
+                elif new_primitives[-1].end.point_distance(primitive.start) < tol:
+                    new_primitives.append(primitive)
+                    primitives.remove(primitive)
+                elif new_primitives[0].start.point_distance(primitive.end) < tol:
+                    new_primitives.insert(0, primitive)
+                    primitives.remove(primitive)
+                elif new_primitives[-1].end.point_distance(primitive.end) < tol:
+                    new_primitives.append(primitive.reverse())
+                    primitives.remove(primitive)
+
+        ordered_wire.primitives = new_primitives
```

Comment:
Better to instantiate a new one

Reply:
Since we are in Wire, we don't know if it is a **2D** or **3D** wire, 
so, I think, I should create `order_primitives` in Wire, and `order_wire` in both **Wire2D** and **Wire3D** ??